Project docker setup and fix `@opensearch-project/opensearch/` node module error (#12)
opensearch-project/automation-app: docker/README.md

> # GitHub Automation App Setup with Docker

Reviewer: Please add a section in the main README under `deployment/docker` heading, and link to this docker README for easier access.

> This GitHub Automation App uses Docker and Docker Compose to build and run a Node.js code with configurable resource and operation settings. Multiple services can be run simultaneously using different configurations.

Reviewer: ```suggestion
The GitHub Automation App can be deployed using Docker and Docker Compose to run as a service with configurable resource and operation settings. Multiple services can be run simultaneously using different configurations.
```

## Prerequisites

Make sure the following installed on the system:

- Docker: [Get Docker](https://docs.docker.com/get-docker/)
- Docker Compose: [Get Docker Compose](https://docs.docker.com/compose/install/)

## Project Structure

> ```bash

Reviewer: Seems like missing the `call` dir here.

.
├── configs/
│   ├── operations/
│   │   ├── github-merged-pulls-monitor.yml

> │   │   └── github-workflow-runs-monitor.yml

Reviewer: Probably remove these specific names and add sample names like the `resources` one.

│   └── resources/
│       └── sample-resource.yml
├── docker/
│   ├── Dockerfile
│   └── docker-compose.yml
├── package.json
└── src/
    └── app.js
```

## Docker Setup

The `docker-compose.yml` is configured to use a Node.js image and to run the app. This mounts the project directory to the container for live reloading.

### Dockerfile

The `Dockerfile` is used to create a Docker image for the app:

```
FROM node:20.17.0
WORKDIR /usr/app
COPY ../package*.json ./
RUN npm cache clean --force

> COPY ../ .

Reviewer: Should link to the dockerfile directly instead of copy paste here.

```

### Docker Compose File

The `docker-compose.yml` file sets up a service (automation-app) to run the app:

```
services:
  automation-app:
    build:
      context: ../
      dockerfile: docker/Dockerfile
    command:
      - /bin/bash
      - -c
      - |
        npm install
        npm start
    environment:
      - RESOURCE_CONFIG=configs/resources/sample-resource.yml
      - OPERATION_CONFIG=${OPERATION_CONFIG}
    volumes:
      - ../:/usr/app/
      - /usr/app/node_modules
    ports:
      - "${PORT}:3000"

> 

Reviewer: Same as above.

```

> ### Running the Services

Reviewer: ```suggestion
### Run multiple Services
```

This allows to run multiple instances of the service with different configurations and ports.

#### This will run the service on port 8080 with the github-merged-pulls-monitor.yml configuration.

```bash
PORT=8080 OPERATION_CONFIG=configs/operations/github-merged-pulls-monitor.yml docker-compose -p automation-app-1 up -d
```

#### This will run the second service on port 8081 with the github-workflow-runs-monitor.yml configuration.

```
PORT=8081 OPERATION_CONFIG=configs/operations/github-workflow-runs-monitor.yml docker-compose -p automation-app-2 up -d

> ```

Reviewer: Use the sample configs and calls.
